sparkle/src/components/Tab.tsx

```diff
@@ -85,12 +90,21 @@ const tabSizingClasses = {
   expand: "s-flex-1",
 };
 
-export function Tab({
+/**
+ * Tab component displaying a list of tabs.
+ *
+ * Tab click action is specified:
+ * - either per tab, by providing either an href or an onClick handler in tab
+ *   data (mutually exclusive);
+ * - or globally, by providing a setCurrentTab function.
+ * Both per-tab and global actions can be set, the per-tab action taking precedence.
```

Comment:
Any reason for this hierarchy?

Reply:
yes, the other way round is less flexible. Here, setCurrentTab can act as "default behaviour" and you can override for a few tab by providing onclick. Other way round, if you provide setCurrentTab, onClick is useless